applications/RANSApplication/custom_strategies/bossak_scalar_scheme.h

```diff
@@ -0,0 +1,371 @@
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+/// A scheme for steady and dynamic equations, using Bossak time integration.
+/**
+ * It can be used for either first- or second-order time derivatives. Elements
+ * and conditions must provide a specialization of SchemeExtension via
+ * their data value container, which allows the scheme to operate independently
+ * of the variable arrangements in the element or condition.
+ */
+template <class TSparseSpace, class TDenseSpace>
+class BossakScalarScheme
+    : public SteadyScalarScheme<TSparseSpace, TDenseSpace>
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    KRATOS_CLASS_POINTER_DEFINITION(BossakScalarScheme);
+
+    using BaseType = SteadyScalarScheme<TSparseSpace, TDenseSpace>;
+
+    using SystemMatrixType = typename BaseType::TSystemMatrixType;
+
+    using SystemVectorType = typename BaseType::TSystemVectorType;
+
+    using LocalSystemVectorType = typename BaseType::LocalSystemVectorType;
+
+    using LocalSystemMatrixType = typename BaseType::LocalSystemMatrixType;
+
+    using DofsArrayType = typename BaseType::DofsArrayType;
+
+    using NodeType = ModelPart::NodeType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor.
+
+    BossakScalarScheme(
+        const double AlphaBossak,
+        const double RelaxationFactor,
+        const Variable<double>& rScalarVariable,
+        const Variable<double>& rScalarRateVariable,
+        const Variable<double>& rRelaxedScalarRateVariable)
+    : BaseType(RelaxationFactor),
+        mAlphaBossak(AlphaBossak),
+        mrScalarVariable(rScalarVariable),
+        mrScalarRateVariable(rScalarRateVariable),
+        mrRelaxedScalarRateVariable(rRelaxedScalarRateVariable)
+    {
+        // Allocate auxiliary memory.
+        const int num_threads = OpenMPUtils::GetNumThreads();
+
+        mMassMatrix.resize(num_threads);
+        mSecondDerivativeValuesVector.resize(num_threads);
+        mSecondDerivativeValuesVectorOld.resize(num_threads);
+    }
+
+    /// Destructor.
+    ~BossakScalarScheme() override = default;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void InitializeSolutionStep(
+        ModelPart& rModelPart,
+        SystemMatrixType& rA,
+        SystemVectorType& rDx,
+        SystemVectorType& rb) override
+    {
+        KRATOS_TRY;
+
+        BaseType::InitializeSolutionStep(rModelPart, rA, rDx, rb);
+
+        const double delta_time = rModelPart.GetProcessInfo()[DELTA_TIME];
+
+        KRATOS_ERROR_IF(delta_time < std::numeric_limits<double>::epsilon())
+            << "detected delta_time = 0 in the Bossak Scheme ... "
+               "check if the time step is created correctly for "
+               "the current model part.";
+
+        BossakScalarScheme::CalculateBossakConstants(
+            mBossak, mAlphaBossak, delta_time);
+
+#pragma omp critical
+        {
+            rModelPart.GetProcessInfo()[BOSSAK_ALPHA] = mBossak.Alpha;
+        }
```

Comment:
I think that you can remove the OpenMP pragma by using the `SetValue`. Indeed, this shouldn't be called within a parallel region so I think that the pragma is not required (maybe I'm wrong).